internal/transport/server_stream.go

```diff
@@ -34,9 +34,10 @@ import (
 type ServerStream struct {
 	*Stream // Embed for common stream functionality.
 
-	st      internalServerTransport
-	ctxDone <-chan struct{}    // closed at the end of stream.  Cache of ctx.Done() (for performance)
-	cancel  context.CancelFunc // invoked at the end of stream to cancel ctx.
+	st                  internalServerTransport
+	ctxDone             <-chan struct{}    // closed at the end of stream.  Cache of ctx.Done() (for performance)
+	cancel              context.CancelFunc // invoked at the end of stream to cancel ctx.
+	deadlineTimerCancel func()             // Invoked at the end of stream.
```

Comment:
Do we always call `cancel`?  Can we piggyback the timer cancellation with it if so?